packages/order-utils/src/asset_balance_and_proxy_allowance_fetcher.ts

```diff
@@ -0,0 +1,101 @@
+import { ERC20ProxyContract, ERC20TokenContract, ERC721ProxyContract, ERC721TokenContract } from '@0x/abi-gen-wrappers';
+import { BigNumber } from '@0x/utils';
+import { BlockParamLiteral, SupportedProvider } from 'ethereum-types';
+import * as _ from 'lodash';
+
+import { AbstractBalanceAndProxyAllowanceFetcher } from './abstract/abstract_balance_and_proxy_allowance_fetcher';
+import { assetDataUtils } from './asset_data_utils';
+
+const UNLIMITED_ALLOWANCE_IN_BASE_UNITS = new BigNumber(2).pow(256).minus(1); // tslint:disable-line:custom-no-magic-numbers
+
+export class AssetBalanceAndProxyAllowanceFetcher implements AbstractBalanceAndProxyAllowanceFetcher {
+    private readonly _stateLayer: BlockParamLiteral;
+    private readonly _erc20Proxy: ERC20ProxyContract;
+    private readonly _erc721Proxy: ERC721ProxyContract;
+    private readonly _provider: SupportedProvider;
+    constructor(
+        erc20Proxy: ERC20ProxyContract,
+        erc721Proxy: ERC721ProxyContract,
+        provider: SupportedProvider,
+        stateLayer: BlockParamLiteral,
+    ) {
+        this._erc20Proxy = erc20Proxy;
+        this._erc721Proxy = erc721Proxy;
+        this._stateLayer = stateLayer;
+        this._provider = provider;
+    }
+    public async getBalanceAsync(assetData: string, userAddress: string): Promise<BigNumber> {
+        const decodedAssetData = assetDataUtils.decodeAssetDataOrThrow(assetData);
```

Comment:
We should be able to refactor this to use DevUtils.

```
devUtils.getBalance(owner, assetData)
```

Which handles ERC20, ERC721, ERC1155 and MAP.